ci-operator/step-registry/cucushift/installer/check/gcp/user-tags/cucushift-installer-check-gcp-user-tags-commands.sh

```diff
@@ -29,46 +29,122 @@ then
   gcloud config set project "${GOOGLE_PROJECT_ID}"
 fi
 
+validation_result_file=$(mktemp)
+
 # User-defined tags validation. It will check if each user-defined tag is applied. 
 # Return non-zero is one or more user-defined tag absent. 
 # $1 - the current tags of the resource under question
 function validate_user_tags() {
   local -r current_tags="$1";  shift
 
-  local ret=0 cnt=1 a_tag_value
+  local cnt=1 a_tag_value
+  echo "" > "${validation_result_file}"
   printf '%s' "${USER_TAGS:-}" | while read -r PARENT KEY VALUE || [ -n "${PARENT}" ]
   do
     a_tag_value="namespacedTagValue: ${PARENT}/${KEY}/${VALUE}"
     if echo "${current_tags}" | grep -Fq "${a_tag_value}"; then
       echo "$(date -u --rfc-3339=seconds) - Found tag ${cnt} '${a_tag_value}' (PARENT/KEY/VALUE)."
       cnt=$(( $cnt + 1 ))
+      echo 0 >> "${validation_result_file}"
       continue
     else
       echo "$(date -u --rfc-3339=seconds) - Failed to find tag '${a_tag_value}' (PARENT/KEY/VALUE)."
-      ret=1
+      echo 1 >> "${validation_result_file}"
     fi
   done
-  return $ret
+}
+
+# check if OCP version will be equal to or greater than the minimum version
+# $1 - the minimum version to be compared with
+# return 0 if OCP version >= the minimum version, otherwise 1
+function version_check() {
+  local -r minimum_version="$1"
+
+  dir=$(mktemp -d)
+  pushd "${dir}"
+
+  cp ${CLUSTER_PROFILE_DIR}/pull-secret pull-secret
+  KUBECONFIG="" oc registry login --to pull-secret
+  ocp_version=$(oc adm release info --registry-config pull-secret ${TESTING_RELEASE_IMAGE} --output=json | jq -r '.metadata.version' | cut -d. -f 1,2)
+
+  echo "[DEBUG] minimum OCP version: '${minimum_version}'"
+  echo "[DEBUG] current OCP version: '${ocp_version}'"
+  curr_x=$(echo "${ocp_version}" | cut -d. -f1)
+  curr_y=$(echo "${ocp_version}" | cut -d. -f2)
+  min_x=$(echo "${minimum_version}" | cut -d. -f1)
+  min_y=$(echo "${minimum_version}" | cut -d. -f2)
+
+  if [ ${curr_x} -ge ${min_x} ] && [ ${curr_y} -ge ${min_y} ]; then
+    echo "[DEBUG] version_check result: ${ocp_version} >= ${minimum_version}"
+    ret=0
+  else
+    echo "[DEBUG] version_check result: ${ocp_version} < ${minimum_version}"
+    ret=1
+  fi
+
+  rm pull-secret
```

Comment:
If any of lines between 70 to 83 failed, the `rm` will be skipped.  Perhaps move the `rm` to line after 68 ?

Reply:
@liangxia The lines between 70 and 83 don't have much things, and I don't think any will fail.